### PR TITLE
Add guidance for using heading classes and update examples

### PR DESCRIPTION
**Pull request by edwardhorsford**
## Description
Per discussions on Slack, this adds guidance for which heading classes to use. I started with the guidance from GOV.UK, and have then modified after several rounds of discussions.

## Why do this?
NHSUK-Frontend currently styles headings by default - `nhsuk-heading-xl` for `h1s`, and `nhsuk-heading-l` for `h2s`. This results in very large headings which may not be suitable on transactional / shorter pages.

Service teams should choose appropriate heading sizes to visually balance on the page, but this `nhsuk-frontend` default likely means that many are using larger headings than they might choose otherwise, thinking it is recommended. GOV.UK faced a similar issue some years ago (not defaulted in code, but all examples used `govuk-heading-xl`), and added explicit guidance for headings and which is most appropriate.

Updating `nhsuk-frontend` to not default sizes or to change the default may be something to investigate in the future. In the short term we can add guidance for teams to suggest what sizes are appropriate.

## Screenshot:

Screenshot 2024-06-07 at 11 56 20

## Checklist

- [ ] Tested against the [NHS.UK frontend testing policy](https://github.com/nhsuk/nhsuk-frontend/blob/main/docs/contributing/testing.md) (Resolution, Browser & Accessibility)
- [x] Code follows the [NHS.UK frontend coding standards](https://github.com/nhsuk/nhsuk-frontend/blob/main/docs/contributing/coding-standards.md)
- [ ] Version number has been increased in `package.json` (using [SEMVER](https://semver.org/))
- [x] CHANGELOG entry
- [ ] [Whats new page](https://service-manual.nhs.uk/whats-new) entry

### CODE REVIEW

**Comment by edwardhorsford** [comment]
Example of a question page before this change:
heading-xl

Example after change:
heading-l

File: app/views/design-system/styles/typography/index.njk [hunk 1]
```diff
@@ -72,13 +72,29 @@
...
+
   {{ designExample({
     group: "styles",
     item: "typography",
-    type: "headings",
+    type: "headings-l",
     htmlOnly: true
   }) }}
 
+  <p>If your page has lots of long form content, start with <code>nhsuk-heading-xl</code> for an <code>{{ "<h1>" | escape }}</code>, <code>nhsuk-heading-l</code> for an <code>{{ "<h2>" | escape }}</code>, and so on.</p>
```
Comment (frankieroberto) [inline app/views/design-system/styles/typography/index.njk:88]: Perhaps "long form content" is a bit vague here?

Could it be something like "If your page is very long and needs 4 heading levels"? And then include an example with an h4 using `nhs-heading-s`?

We could also mention that even on short pages, you might still want the h1 to use XL but the h2 to use M, eg if the h1 title is very short? 🤔

Comment (edwardhorsford) [reply app/views/design-system/styles/typography/index.njk:90]: Even with a short heading don't think we'd recommend xl.

'long form content' comes from how the GOV.UK design system words it. I'm not sure that's the same as 'very long' - it's more about it being primarily content based.

**Review by frankieroberto — APPROVED** [review]

File: app/views/design-system/styles/typography/index.njk [hunk 2]
```diff
@@ -72,10 +72,29 @@
...
+    group: "styles",
+    item: "typography",
+    type: "headings-l",
+    htmlOnly: true
+  }) }}
+
+  <h3>Other types of pages</h3>
+
+  <p>Some pages may benefit from a larger <code>{{ "<h1>" | escape }}</code> to allow for better visual balance and more heading levels. Examples include longer content pages, start pages and dashboards. For these pages, start with <code>nhsuk-heading-xl</code> for an <code>{{ "<h1>" | escape }}</code>, <code>nhsuk-heading-l</code> for <code>{{ "<h2>" | escape }}</code>, and so on.</p>
+
```
Comment (frankieroberto) [inline app/views/design-system/styles/typography/index.njk:93]: Could add something to suggest that you can also skip typography levels if needed?

```suggestion
<p>You can also skip typography sizes if needed, but should not skip heading levels. For example you could use <code>nhsuk-heading-l</code> for an <code>{{ "<h2>" | escape }}</code> and  <code>nhsuk-heading-xs</code> for an <code>{{ "<h3>" | escape }}</code>.</p>
```

File: app/views/design-system/styles/typography/index.njk [hunk 3]
```diff
@@ -72,10 +72,29 @@
...
+  {{ designExample({
+    group: "styles",
+    item: "typography",
+    type: "headings-l",
+    htmlOnly: true
+  }) }}
+
+  <h3>Other types of pages</h3>
+
+  <p>Some pages may benefit from a larger <code>{{ "<h1>" | escape }}</code> to allow for better visual balance and more heading levels. Examples include longer content pages, start pages and dashboards. For these pages, start with <code>nhsuk-heading-xl</code> for an <code>{{ "<h1>" | escape }}</code>, <code>nhsuk-heading-l</code> for <code>{{ "<h2>" | escape }}</code>, and so on.</p>
```
Comment (sarawilcox) [inline app/views/design-system/styles/typography/index.njk:92]:
```suggestion
  <p>Some pages benefit from a larger <code>{{ "<h1>" | escape }}</code> to allow for better visual balance and more heading levels. Examples include:<p>
  <ul>
  <li>longer content pages</li>
  <li>start pages</li>
  <li>dashboards</li>
  <p>For these pages, start with <code>nhsuk-heading-xl</code> for the <code>{{ "<h1>" | escape }}</code>, <code>nhsuk-heading-l</code> for <code>{{ "<h2>" | escape }}</code>, and so on.</p>
```

File: app/views/design-system/styles/typography/index.njk [hunk 4]
```diff
@@ -72,10 +72,29 @@
...
+  <p>Shorter pages, such as question pages, should use <code>nhsuk-heading-l</code> for main page headings <code>{{ "<h1>" | escape }}</code>, followed by <code>nhsuk-heading-m</code> for <code>{{ "<h2>" | escape }}</code> and so on.</p>
+
+  {{ designExample({
+    group: "styles",
+    item: "typography",
+    type: "headings-l",
+    htmlOnly: true
+  }) }}
+
+  <h3>Other types of pages</h3>
```
Comment (sarawilcox) [inline app/views/design-system/styles/typography/index.njk:90]:
```suggestion
  <h3>Headings in long content and other types of pages</h3>
```

File: app/views/design-system/styles/typography/index.njk [hunk 5]
```diff
@@ -72,10 +72,29 @@
 
   <h2 id="headings">Headings</h2>
 
+  <p>Use heading tags, such as <code>{{ "<h1>" | escape }}</code>, <code>{{ "<h2>" | escape }}</code> and so on, to tag the headings on a page. Apply a heading class, such as <code>nhsuk-heading-l</code>, to style them visually. Style headings consistently to create a clear content structure throughout your service.</p>
+
+  <p>Write all headings in sentence case.</p>
+
+  <h3>Question pages and shorter content pages</h3>
+
+  <p>Shorter pages, such as question pages, should use <code>nhsuk-heading-l</code> for main page headings <code>{{ "<h1>" | escape }}</code>, followed by <code>nhsuk-heading-m</code> for <code>{{ "<h2>" | escape }}</code> and so on.</p>
```
Comment (sarawilcox) [inline app/views/design-system/styles/typography/index.njk:81]:
```suggestion
  <p>For question pages and short content pages across a transactional journey, use <code>nhsuk-heading-l</code> for main page headings <code>{{ "<h1>" | escape }}</code>, followed by <code>nhsuk-heading-m</code> for <code>{{ "<h2>" | escape }}</code> and so on.</p>
```

File: app/views/design-system/styles/typography/index.njk [hunk 6]
```diff
@@ -72,10 +72,29 @@
 
   <h2 id="headings">Headings</h2>
 
+  <p>Use heading tags, such as <code>{{ "<h1>" | escape }}</code>, <code>{{ "<h2>" | escape }}</code> and so on, to tag the headings on a page. Apply a heading class, such as <code>nhsuk-heading-l</code>, to style them visually. Style headings consistently to create a clear content structure throughout your service.</p>
+
+  <p>Write all headings in sentence case.</p>
+
+  <h3>Question pages and shorter content pages</h3>
```
Comment (sarawilcox) [inline app/views/design-system/styles/typography/index.njk:79]:
```suggestion
  <h3>Headings in transactional journeys</h3>
```

**Review by sarawilcox — COMMENTED** [review]
Hope I've done this correctly. Have suggested changes.

File: app/views/design-system/styles/typography/index.njk (hunk 2 above)
Comment (edwardhorsford) [reply app/views/design-system/styles/typography/index.njk:98]: @anandamaryon1 @sarawilcox any thoughts on this?

**Comment by sarawilcox** [comment]
https://github.com/nhsuk/nhsuk-service-manual/pull/1974#discussion_r1715544450

I think the above looks fine. I am looking at adding some guidance around not going down to a heading level 5 in terms of content structure. (That suggests that content is too complex.)

**Comment by sarawilcox** [comment]
In branch https://github.com/nhsuk/nhsuk-service-manual/tree/review/1-release-7-0-0 awaiting final sign offs and release.

**Comment by anandamaryon1** [comment]
Released as part of v7.0.0: https://github.com/nhsuk/nhsuk-service-manual/pull/2040

**Pull request closed without merging**